ble_gatt_chr_def missing `cpfd` on ESP-IDF version 5.2.1

```
error[E0063]: missing field `cpfd` in initializer of `ble_gatt_chr_def`
  --> C:\Users\ryankopf\.cargo\registry\src\index.crates.io-6f17d22bba15001f\esp32-nimble-0.6.0\src\server\ble_service.rs:52:19
   |
52 |             .push(esp_idf_sys::ble_gatt_chr_def {
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `cpfd`

For more information about this error, try `rustc --explain E0063`.
error: could not compile `esp32-nimble` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

It seems a new field has been added to this struct. I got around it by adding `cpfd: core::ptr::null_mut(),` but I'm sure this is not ideal in the long run.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -21,22 +21,22 @@ jobs:
             idf-version: release/v4.4
             rustflags: "-C default-linker-libraries"
           - target: riscv32imc-esp-espidf
-            idf-version: v5.1.2
+            idf-version: v5.1.3
             rustflags: "-C default-linker-libraries --cfg espidf_time64"
           - target: riscv32imc-esp-espidf
-            idf-version: v5.2
+            idf-version: v5.2.1
             rustflags: "-C default-linker-libraries --cfg espidf_time64"
           - target: riscv32imac-esp-espidf
-            idf-version: v5.1.2
+            idf-version: v5.1.3
             rustflags: "-C default-linker-libraries --cfg espidf_time64"
             mcu: esp32c6
           - target: xtensa-esp32-espidf
             idf-version: release/v4.4
           - target: xtensa-esp32-espidf
-            idf-version: v5.1.2
+            idf-version: v5.1.3
             rustflags: "--cfg espidf_time64"
           - target: xtensa-esp32-espidf
-            idf-version: v5.2
+            idf-version: v5.2.1
             rustflags: "--cfg espidf_time64"
 
     steps:

--- a/src/server/ble_characteristic.rs
+++ b/src/server/ble_characteristic.rs
@@ -97,6 +97,12 @@ pub struct BLECharacteristic {
   svc_def_descriptors: Vec<esp_idf_sys::ble_gatt_dsc_def>,
   subscribed_list: Vec<(u16, NimbleSub)>,
   on_subscribe: Option<Box<dyn FnMut(&Self, &BLEConnDesc, NimbleSub) + Send + Sync>>,
+  #[cfg(all(
+    esp_idf_version_major = "5",
+    esp_idf_version_minor = "2",
+    not(esp_idf_version_patch = "0")
+  ))]
+  pub(crate) cpfd: Option<esp_idf_sys::ble_gatt_cpfd>,
 }
 
 impl BLECharacteristic {
@@ -113,6 +119,12 @@ impl BLECharacteristic {
       svc_def_descriptors: Vec::new(),
       subscribed_list: Vec::new(),
       on_subscribe: None,
+      #[cfg(all(
+        esp_idf_version_major = "5",
+        esp_idf_version_minor = "2",
+        not(esp_idf_version_patch = "0")
+      ))]
+      cpfd: None,
     }
   }
 
@@ -246,6 +258,16 @@ impl BLECharacteristic {
         unsafe { esp_idf_sys::ble_gattc_notify_custom(it.0, self.handle, om) };
       }
     }
+  }
+
+  #[cfg(all(
+    esp_idf_version_major = "5",
+    esp_idf_version_minor = "2",
+    not(esp_idf_version_patch = "0")
+  ))]
+  /// Set the Characteristic Presentation Format.
+  pub fn cpfd(&mut self, cpfd: esp_idf_sys::ble_gatt_cpfd) {
+    self.cpfd = Some(cpfd);
   }
 
   pub(super) extern "C" fn handle_gap_event(

--- a/src/server/ble_characteristic.rs
+++ b/src/server/ble_characteristic.rs
@@ -102,7 +102,7 @@ pub struct BLECharacteristic {
     esp_idf_version_minor = "2",
     not(esp_idf_version_patch = "0")
   ))]
-  pub(crate) cpfd: Option<esp_idf_sys::ble_gatt_cpfd>,
+  pub(crate) cpfd: [esp_idf_sys::ble_gatt_cpfd; 2],
 }
 
 impl BLECharacteristic {
@@ -124,7 +124,7 @@ impl BLECharacteristic {
         esp_idf_version_minor = "2",
         not(esp_idf_version_patch = "0")
       ))]
-      cpfd: None,
+      cpfd: [Default::default(); 2],
     }
   }
 
@@ -267,7 +267,7 @@ impl BLECharacteristic {
   ))]
   /// Set the Characteristic Presentation Format.
   pub fn cpfd(&mut self, cpfd: esp_idf_sys::ble_gatt_cpfd) {
-    self.cpfd = Some(cpfd);
+    self.cpfd[0] = cpfd;
   }
 
   pub(super) extern "C" fn handle_gap_event(

--- a/src/server/ble_service.rs
+++ b/src/server/ble_service.rs
@@ -57,6 +57,15 @@ impl BLEService {
               flags: chr.properties.bits(),
               min_key_size: 0,
               val_handle: &mut chr.handle,
+              #[cfg(all(
+                esp_idf_version_major = "5",
+                esp_idf_version_minor = "2",
+                not(esp_idf_version_patch = "0")
+              ))]
+              cpfd: match chr.cpfd.as_mut() {
+                Some(cpfd) => cpfd,
+                None => core::ptr::null_mut(),
+              },
             });
         }
         self

--- a/src/server/ble_service.rs
+++ b/src/server/ble_service.rs
@@ -62,10 +62,7 @@ impl BLEService {
                 esp_idf_version_minor = "2",
                 not(esp_idf_version_patch = "0")
               ))]
-              cpfd: match chr.cpfd.as_mut() {
-                Some(cpfd) => cpfd,
-                None => core::ptr::null_mut(),
-              },
+              cpfd: chr.cpfd.as_mut_ptr(),
             });
         }
         self

